applications/FluidDynamicsApplication/custom_elements/surface_smoothing_element.cpp

```diff
@@ -0,0 +1,670 @@
+                    // Store the Gauss points weights
+                    const double face_weight = face_jacobians(i_gauss) * face_gauss_pts[i_gauss].Weight();
+
+                    // Compute the global coordinates of the face Gauss pt.
+                    GeometryType::CoordinatesArrayType global_coords = ZeroVector(num_dim);
+                    global_coords = outer_face.GlobalCoordinates(global_coords, face_gauss_pts[i_gauss].Coordinates());
+
+                    // Compute the parent geometry local coordinates of the Gauss pt.
+                    GeometryType::CoordinatesArrayType loc_coords = ZeroVector(num_dim);
+                    loc_coords = GetGeometry().PointLocalCoordinates(loc_coords, global_coords);
+
+                    // Compute shape function values
+                    // Obtain the parent subgeometry shape function values
+                    Kratos::Vector face_shape_func;
+                    face_shape_func = GetGeometry().ShapeFunctionsValues(face_shape_func, loc_coords);
+
+                    double temp_value = 0.0;
+
+                    for (unsigned int i = 0; i < num_nodes; i++){
+
+                        VectorType solid_normal = ZeroVector(num_dim);
+                        if (GetGeometry()[i].GetValue(IS_STRUCTURE) == 1.0)
+                        {
+                            solid_normal = GetGeometry()[i].FastGetSolutionStepValue(NORMAL);
+                            const double norm = Kratos::norm_2(solid_normal);
+                            solid_normal = (1.0/norm)*solid_normal;
+                        }
+
+                        temp_value += face_shape_func(i)*Kratos::inner_prod(solid_normal, GradPHIold[i]);
+                    }
+
+                    for (unsigned int i = 0; i < num_nodes; i++){
+                        tempBCRHS[i] += epsilon * temp_value * face_weight * face_shape_func(i);
+                    }
+                }
+            }
+
+        }
+    }
+
+    noalias(rLeftHandSideMatrix) = tempM + tempA;
+    noalias(rRightHandSideVector) = tempBCRHS + prod(tempM,PHIold) - prod(rLeftHandSideMatrix,PHIdof);
+
+    KRATOS_CATCH("");
+}
+
+/**
+ * this is called during the assembling process in order
+ * to calculate the elemental left hand side matrix only
+ * @param rLeftHandSideMatrix: the elemental left hand side matrix
+ * @param rCurrentProcessInfo: the current process info instance
+ */
+void SurfaceSmoothingElement::CalculateLeftHandSide(MatrixType& rLeftHandSideMatrix, ProcessInfo& rCurrentProcessInfo)
+{
+}
+
+/**
+ * this is called during the assembling process in order
+ * to calculate the elemental right hand side vector only
+ * @param rRightHandSideVector: the elemental right hand side vector
+ * @param rCurrentProcessInfo: the current process info instance
+ */
+void SurfaceSmoothingElement::CalculateRightHandSide(VectorType& rRightHandSideVector, ProcessInfo& rCurrentProcessInfo)
+{
+}
```

Comment:
If you are intentionally not implementing this, I'd throw an error in case it is called.

Reply:
I will rely on the base element.